### PR TITLE
RD-2664 Fix selecting label key from dropdown list in filters_spec

### PR DESCRIPTION
**Pull request by qooban**
I found out that the only difference between label key selecting function (`selectRuleLabelKey`) and other functions setting values in dropdowns in `filters_spec` (`selectRuleRowType`, `selectRuleOperator`, `selectRuleAttributeValues`, `selectRuleLabelValues`) was that in `selectRuleLabelKey` there was no `click` issued on the dropdown's div element. I suspect that might be an issue.

This PR contains two changes:
1. https://github.com/cloudify-cosmo/cloudify-stage/pull/1519/commits/741bbec6435cbe1442e6e6ca57beb3bd426ade98 - small change, just `click` addition in `selectRuleLabelKey` (that should fix the problem)
2. https://github.com/cloudify-cosmo/cloudify-stage/pull/1519/commits/8ecc3ae844fc454b09ec87fe6f195d806b0f61d8 - refactoring in dropdowns usage inside `filters_spec`. The goal was to reuse as much dropdown interaction code as possible. There is much more to do within that area IMHO (many other tests use their own logic to interact with dropdowns), so I have created RD-2829 to create generic Cypress custom commands for interaction with dropdowns.

### System Tests

1. [Stage-UI-System-Test #971](https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/971/pipeline) - only `filters_spec` - PASSED
2. [Stage-UI-System-Test #972](https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/972/pipeline) - only `filters_spec` - PASSED
3. [Stage-UI-System-Test #973](https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/973/pipeline) - only `filters_spec` - PASSED
4. [Stage-UI-System-Test #976](https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/976/pipeline) - full run - FAILED (Internal Server Error 500 during EC2 instance creation)
5. [Stage-UI-System-Test #977](https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/977/pipeline) - full run - FAILED (Internal Server Error 500 during EC2 instance creation)
6. [Stage-UI-System-Test #978](https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/978/pipeline) - full run - FAILED (Internal Server Error 500 during some test execution, but `filters_spec` test passed)
7. [Stage-UI-System-Test #980](https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/980/pipeline) - full run - FAILED (Internal Server Error 500 during EC2 instance creation)
8. [Stage-UI-System-Test #981](https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/981/pipeline) - full run - FAILED (Internal Server Error 500 during EC2 instance creation)
9. [Stage-UI-System-Test #982](https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/982/pipeline) - full run - FAILED (Internal Server Error 500 during some test execution, but `filters_spec` test passed)

### CODE REVIEW

**Comment by Gelio** [comment]
Looks promising :slightly_smiling_face: I'll review the PR tomorrow

File: test/cypress/integration/widgets/filters_spec.ts [hunk 1]
```diff
@@ -51,8 +51,13 @@ describe('Filters widget', () => {
         cy.contains('Add new rule').click();
     }
 
+    type FilterRuleDropdownNames = 'ruleRowType' | 'ruleOperator' | 'ruleValue' | 'labelKey' | 'labelValue';
+    function openDropdown(divName: FilterRuleDropdownNames) {
```
Comment (Gelio) [inline test/cypress/integration/widgets/filters_spec.ts:55]: Minor:

Seeing the type pluralized here does not sound natural :smile: It leads me to think that it should be singular (`FilterRuleDropdownName`), similarly to e.g. `divName: string` and not `divName: strings` :slightly_smiling_face:

File: test/cypress/integration/widgets/filters_spec.ts [hunk 2]
```diff
@@ -399,21 +403,36 @@ describe('Filters widget', () => {
...
+        }
+
+        type RuleValueObject = { value: string; isNew: boolean };
+        function searchAndSetDropdownValue(
+            { value, isNew }: RuleValueObject,
+            searchEndpoint: string | RegExp,
+            additionLabel: string,
+            isMultiple: boolean
+        ) {
+            if (searchEndpoint) cy.interceptSp('GET', searchEndpoint).as(`search_${value}`);
```
Comment (Gelio) [inline test/cypress/integration/widgets/filters_spec.ts:417]: Minor:

What's the reason to have an `if` if `searchEndpoint` is mandatory? :thinking: Same with other `if (searchEndpoint)`

I see you're using `''` to disable this functionality. Maybe instead we should add `undefined` as a possible value for `searchEndpoint`?

**Review by Gelio — APPROVED** [review]
Cool refactors, looks good to me :+1: 

I've left 2 minor comments :smile:

File: test/cypress/integration/widgets/filters_spec.ts (hunk 1 above)
Comment (qooban) [reply test/cypress/integration/widgets/filters_spec.ts]: My mistake, not intentional. Fixed 🙂

File: test/cypress/integration/widgets/filters_spec.ts (hunk 2 above)
Comment (qooban) [reply test/cypress/integration/widgets/filters_spec.ts:417]: Good catch. I moved `searchEndpoint` to be the last argument and to be optional. Updated `searchAndSetDropdownValue` calls accordingly and used `undefined` in one case when we don't want to check searching requests.

Comment (Gelio) [reply test/cypress/integration/widgets/filters_spec.ts:417]: I thought that making it accept `string | RegExp | undefined` would have been sufficient :smile: But making it be optional is good enough :slightly_smiling_face:

**Review by Gelio — APPROVED** [review]
LGTM :+1: Thanks for addressing the comments :smile:

**Comment by qooban** [comment]
@stepek Let me know if you plan to review this PR. I'd like to merge it to `master` today if possible and then cherry-pick it to `6.1-build` branch.

**Review by stepek — APPROVED** [review]
LGTM!

**Pull request merged**